src/sentry/tasks/groupowner.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import absolute_import, print_function
+
+import logging
+
+from datetime import timedelta
+
+from django.utils import timezone
+
+from sentry.eventstore.models import Event
+from sentry.eventstore.processing import event_processing_store
+from sentry.models import Project, Release
+from sentry.models.groupowner import GroupOwner, GroupOwnerType
+from sentry.tasks.base import instrumented_task
+from sentry.utils.committers import get_serialized_event_file_committers
+
+PREFERRED_GROUP_OWNERS = 3
+PREFERRED_GROUP_OWNER_AGE = timedelta(days=1)
+
+logger = logging.getLogger("tasks.groupowner")
+
+
+@instrumented_task(
+    name="sentry.tasks.process_suspect_commits", default_retry_delay=5, max_retries=5
+)
+def process_suspect_commits(group_id, cache_key, **kwargs):
+    if not group_id:
+        return
+
+    can_process = True
+    data = event_processing_store.get(cache_key)
+    if not data:
+        logger.info(
+            "process_suspect_commits.skipped",
+            extra={"cache_key": cache_key, "reason": "missing_cache"},
+        )
+        return
+
+    event = Event(
+        project_id=data["project"], event_id=data["event_id"], group_id=group_id, data=data
+    )
+    project = Project.objects.get_from_cache(id=event.project_id)
+    owners = GroupOwner.objects.filter(
```

Comment:
I don't think this query can execute for every event. Since the fast path/early return only needs the count I would put the count + information about the oldest owner into memcached (it could be redis too, talk to james).

To be clear, `get_many_from_cache` can *not* be used here. It can only query by fields that are unique/pk, and group_id (or any of the things you're querying for) is not unique per-GroupOwner.

I _think_ this prevents you from even enabling this featureflag on the `sentry/sentry` project alone fwiw. My guess is that the deploy will succeed but ops will notice.

Reply:
That's a really good suggestion. I put the owner data in the cache with a 60 second timeout...so we won't have to do this calculation every event - only the first 3 per group then once every 60 seconds. This should cut down the work significantly.

I do feel like I also want to somehow also cache the event data so that I only calculate this on events that could result in a different owner.